### PR TITLE
[inductor][XPU] Fix AMP autocast round-trip fp16 accuracy (#188261)

### LINKED ISSUE #188261: [inductor][XPU] PR #184366 causes AMP fp16 training accuracy regression on XPU
### 🐛 Describe the bug

After PR #184366 ("[inductor] Preserve narrow dtype convert chains", commit `6f94c6e16e1bf562f8f5514953cf9b8d47d07cdd`), AMP fp16 training accuracy fails for `fbnetv3_b` (timm) on XPU (BMG).

The change in `torch/_inductor/fx_passes/joint_graph.py` adds:

```python
# A narrower intermediate can be worth materializing before upcasting.
if dtype1.itemsize < dtype2.itemsize:
    return
```

This prevents the `pointless_convert` pass from removing `fp32→fp16→fp32` cast chains, causing precision loss in BatchNorm gradient reductions.

### Reproduction

```bash
# torch 2.14.0.dev20260624+xpu
pip install --pre torch --index-url https://download.pytorch.org/whl/nightly/xpu
pip install timm

# Add "fbnetv3_b 64" to benchmarks/dynamo/timm_models_list.txt, then:
TORCHINDUCTOR_FORCE_DISABLE_CACHES=1 python benchmarks/dynamo/timm_models.py \
  --accuracy -d xpu -n10 --backend=inductor \
  --cold-start-latency --training --amp --amp-dtype float16 \
  --only fbnetv3_b
```

Result:
```
RMSE (res-fp64): 0.00462, (ref-fp64): 0.00073 and shape=torch.Size([16]).
Accuracy failed for key name blocks.0.0.bn1.weight.grad
fail_accuracy
```

### Evidence

**Bisect:**
- Last good nightly: `dev20260602` (commit `e8fdac84fc`)
- First bad nightly: `dev20260603` (commit `7567e9484a`)
- Offending commit: `6f94c6e16e1` (PR #184366)

**Backend isolation:**
- `eager`: PASS
- `aot_eager`: PASS
- `inductor`: FAIL

This confirms the issue is in the inductor optimization pass, not in AOTAutograd or Dynamo tracing.

**Revert verification:**

Removing the 3-line `itemsize` check from `joint_graph.py` (line ~852) together with setting `TORCHINDUCTOR_EMULATE_PRECISION_CASTS_ON_SAVED_TENSORS=0` restores the test to PASS.

Note: `emulate_precision_casts_on_saved_tensors` (PR #184110) blocks the same optimization through a separate code path. Both need to be reverted for the test to pass on the latest nightly (`dev20260624`).

### Versions

- PyTorch: 2.14.0.dev20260624+xpu
- Affected since: dev20260603
- Also present on `release/2.13` branch
- Hardware: Intel Arc B580 (BMG)

cc @mcarilli @ptrblck @leslie-fang-intel @jgong5 @chauhang @penguinwu @voznesenskym @EikanWang @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo @gujinghui @fengyuan14 @guangyey @jansel @eellison

### PR DESCRIPTION
**Pull request by PannenetsF**
Fixes #188261

Collapse redundant AMP autocast dtype round-trip conversions in the Inductor joint graph FX pass to prevent fp16 accuracy loss on XPU. When an operator's input and output are both in fp16 after autocast, the intermediate cast through fp32 is unnecessary and introduces precision loss.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by weishi-deng** [comment]
Hi @PannenetsF Any updates for this fix?

**Comment by weishi-deng** [comment]
@PannenetsF Remainder: This model acc issue is caused by the combination of https://github.com/pytorch/pytorch/pull/184110 and https://github.com/pytorch/pytorch/pull/184366. The changes of only pointless_convert don't help fix.

**Comment by PannenetsF** [comment]
Hi @weishi-deng ， sorry for the missing reply. I did some exps to verify your points. 

I confirmed on the latest nightly (torch 2.15.0.dev20260819) in a CUDA devbox, and it turned to be a PASS. As I have no access to XPU devbox, I cannot continue on this issue. Could you take it if possible? Thanks!
